src/keycloak/openid_connect.py

```diff
@@ -92,9 +92,11 @@ def decode_token(self, token, key, algorithms=None, **kwargs):
         :raises jose.exceptions.JWTClaimsError: If any claim is invalid in any
             way.
         """
+        filtered_kwargs = {k: v for k, v in kwargs.items() if k != 'audience'}
         return jwt.decode(token, key,
                           audience=kwargs.get('audience') or self._client_id,
```

Comment:
```suggestion
                          audience=kwargs.get('audience') or self._client_id,
```
Simply replace `kwargs.get('audience')` with `kwargs.pop('audience', None)` and drop additional filtering for `kwargs` 
#8

Reply:
Nice!